angular-js/src/app/mrz-scanner/mrz-scanner.component.ts

```diff
@@ -62,58 +62,32 @@ export class MrzScannerComponent implements OnInit {
     alert(e.name + ': ' + e.message);
   }
 
+  toConfidenceString(input: any, key: string): string {
+    const confidence = input[key].confidence;
+
+    if (!confidence) {
+      return "";
+    }
+    return ` (${Number(confidence).toFixed(3)})`
+  }
+
+  parseMRZValue(input: any, key: string) {
+    return input[key] ? (input[key].value + this.toConfidenceString(input, key)) : ''
+  }
```

Comment:
It's more of a generic suggestion, but I would suggest using a linter for all the projects. For example if we take these lines I see:

- input is of `any` type
- the function does not specify a return type
- missing semicolon after the return (or redundant semicolons in other places? Depends on what convention you want to follow)

Plus some minor observations:

- if value is undefined the text might be corrupted
- all these methods could be static, and perhaps in a separate class, which would make it easier to duplicate and synchronize between the 3 projects
- `toConfidenceString` should probably take the confidence value as the parameter and return the string value without spaces and parenthesis

Reply:
> all these methods could be static, and perhaps in a separate class, which would make it easier to duplicate and synchronize between the 3 projects

The methods need to be duplicated, because these are completely separate solutions without a convenient way to share code. As such, since it's only one function inside that component, it makes more sense from an "example code" perspective to have it all together (unless the component itself gets too convoluted, then we'll have to separate it)

> toConfidenceString should probably take the confidence value as the parameter and return the string value without spaces and parenthesis

Not all properties have a confidence value -- and in those cases we don't want to display anything in parentheses, even omit the parentheses. I added all of that to one function, else we'd have to have another separate place to handle cases where confidence is missing and whether to add spaces and parentheses.

Everything else I agree with and I'll fix it